qiita_db/support_files/patches/python_patches/41.py

```diff
@@ -0,0 +1,109 @@
+
+from subprocess import Popen, PIPE
+from os.path import exists, join
+
+from qiita_db.sql_connection import TRN
+from qiita_db.artifact import Artifact
+from qiita_db.util import (insert_filepaths, convert_to_id, get_mountpoint,
+                           get_mountpoint_path_by_id)
+
+
+tgz_id = convert_to_id("tgz", "filepath_type")
+_, analysis_mp = get_mountpoint('analysis')[0]
+
+with TRN:
+    #
+    # Generating compressed files for picking failures -- artifact_type = BIOM
+    #
+    sql = """SELECT artifact_id FROM qiita.artifact
+                JOIN qiita.artifact_type USING (artifact_type_id)
+                WHERE artifact_type = 'BIOM'"""
+    TRN.add(sql)
+
+    for r in TRN.execute_fetchindex():
+        to_tgz = None
+        a = Artifact(r[0])
+        for _, fp, fp_type in a.filepaths:
+            if fp_type == 'directory':
+                # removing / from the path if it exists
+                to_tgz = fp[:-1] if fp[-1] == '/' else fp
+                break
+
+        if to_tgz is None:
+            continue
+
+        tgz = to_tgz + '.tgz'
+        if not exists(tgz):
+            cmd = 'tar zcf %s %s' % (tgz, to_tgz)
```

Comment:
Is that a known issue? It just seems unnecessary to use a subprocess, since there's already a module to do this.